lib/widgets-customize.php

```diff
@@ -156,15 +156,20 @@ function gutenberg_customize_widgets_init() {
  * @return bool Filtered decision about loading block assets.
  */
 function gutenberg_widgets_customize_load_block_editor_scripts_and_styles( $is_block_editor_screen ) {
-	if (
-		gutenberg_use_widgets_block_editor() &&
-		is_callable( 'get_current_screen' ) &&
-		'customize' === get_current_screen()->base
-	) {
-		return true;
+	if ( ! gutenberg_use_widgets_block_editor() ) {
+		return $is_block_editor_screen;
+	}
+
+	if ( ! is_callable( 'get_current_screen' ) ) {
+		return $is_block_editor_screen;
+	}
+
+	$current_screen = get_current_screen();
+	if ( is_null( $current_screen ) ) {
```

Comment:
Why not check for the `instanceof WP_Screen`? 
- As this is a plugin, an additional check might be required to ensure the class `WP_Screen` exists. 
- `get_current_screen` returns `null` on failure or an instance of `WP_Screen` on success. 

The check here is for the failure state. Using this approach keeps the check positive and simple, while avoiding an additional `class_exists` check.

Reply:
FWIW, you can safely do an `instanceof` check with a class that does not exist., https://3v4l.org/X4n64.